capstone/src/main/java/com/google/sps/Objects/Post.java

```diff
@@ -46,9 +47,46 @@ public static Post getPostEntity(Entity entity) {
       ? new HashSet<>() 
       : new HashSet<String>((ArrayList<String>) entity.getProperty("likes"));   
     ArrayList<Comment> comments = new ArrayList<>();
+     if (entity.getProperty("comments") != null) {
+      createCommentObjectList(comments, entity);
+    }
     return new Post(postId, authorId, postText, comments, challengeName, timestamp, img, likes);
   }
 
+  private static void createCommentObjectList(ArrayList<Comment> comments, Entity entity) {
+    ArrayList<EmbeddedEntity> commentEntitys = (ArrayList<EmbeddedEntity>) entity.getProperty("comments");
+    for (EmbeddedEntity comment: commentEntitys) {
+      comments.add(Comment.getCommentEntity(comment));
+    }
+  }
+
+  public Entity createPostEntity() {
```

Comment:
I can't comment on line 39 `public static Post getPostEntity(Entity entity)` because you didn't change it, but I think it makes sense to rename that method to something like getFromPostEntity (or just fromEntity) for consistency/clarity